js/hview.js

```diff
@@ -608,57 +615,42 @@ $(function () {
+            $(".slider-group").toggleClass("visible hidden")
+            $("#vlist, #border-box").addClass("multi-page")
+        }
 
         const vListJQueries /* jQuery[] */  = []
         const boxJQueries /* jQuery[] */ = []
-        const rootNodes /* ViewNode[] */ = []
-
-        /* Pre-process all the view hierarchies into ready to use HTML that can be loaded into the 
-           web browser with low latency. Also immediately show the first parsed view hierarchy */
-        for(const [filename, file] of Object.entries(jsZip.files)) {
-            /* The JSZip.files property references subFolders and hidden folders in addition to
-               the desired child folders. These need to be filtered out during preprocessing.
-               Note: This filtering process might not be necessary depending on the format of real data. */
-            if (!(filename.startsWith("_")) && file.dir && !(filename.endsWith("img/"))) {
-
-                const subZip /* JSZip */ = jsZip.folder(filename)
-                const subConfig /* JSONObject or JSONArray */ = JSON.parse(subZip.file("config.json").asText())
-                currentAppInfo /* Object */ = { type: TYPE_ZIP, data: subZip, config: subConfig, name: subConfig.title }
-
-                if (viewController == null) {
-                    viewController = createViewController(currentAppInfo)
-
-                    /* Some UI setup needs to wait to run until after the appInfo / view controller has been initialized. */
-                    setupCustomCommandButton(viewController)
-                    setupWindowTitle(currentAppInfo)
-                }
+        let rootNodes /* ViewNode[] */ = []
 
-                viewController.zip = subZip;
-                viewController.loadViewList().then(rootNode => {
-                    /* renderList attaches the html to the vList container and box container provided.
-                       By providing temporary holder containers, and then storing those after running
-                       renderList(), the UX can be preprocessed and stored for later use. */
-                    const tBox = $("<div>")
-                    const tVList = $("<div>")
-                    renderList(rootNode, tVList, tBox, vListJQueries.length)
-                    vListJQueries.push(tVList)
-                    boxJQueries.push(tBox)
-                    rootNodes.push(rootNode)
-
-                    /* If content isn't yet displayed, show the parsed content */
-                    if ($("#vlist_content label").length == 0) {
-                        tVList.children().appendTo("#vlist_content")
-                        tBox.children().appendTo("#border-box")
-                        applyResizeData();
-                    }
-                }).catch(msg => { handleLoadingListError(msg) });
+        viewController.loadViewList().then(function(nodes) {
+            rootNodes = nodes
+
+            for(let i = 0; i < rootNodes.length; i++) {
```

Comment:
This loop is probably causing ANR in chrome. We can probably add a deferred logic here in a followup cl

Reply:
It certainly is. I'll fix this in a future CL.